soroban-env-host/src/native_contract/token/balance.rs

```diff
@@ -107,16 +106,16 @@ pub fn spend_balance_no_freeze_check(
 
 // Metering: covered by components.
 pub fn spend_balance(e: &Host, id: Identifier, amount: i128) -> Result<(), HostError> {
-    let is_frozen = read_state(e, id.clone())?;
+    let is_frozen = is_deauthorized(e, id.clone())?;
```

Comment:
`is_deauthorized` or something like that? You should probably just do find-and-replace for 'frozen'.

Reply:
Fixed